### PR TITLE
Added MuiButtonBaseProps to DrawerRailItemProps

### PR DESCRIPTION
**Pull request by bkarambe**
Fixes BLUI-3122

#### Changes proposed in this Pull Request:
- Added MuiButtonBaseProps in DrawerRailItemProps

#### To Test:
- yarn start:showcase

### CODE REVIEW

**Comment by jeffvg** [comment]
do you have an example of how to use this for rail?

File: components/src/core/Drawer/DrawerRailItem/DrawerRailItem.tsx [hunk 1]
```diff
@@ -42,48 +42,49 @@ const useUtilityClasses = (ownerState: DrawerRailItemProps): Record<DrawerRailIt
 };
 
 export type ExtendedNavItem = NavItem & { ButtonBaseProps?: Partial<MuiButtonBaseProps> };
-export type DrawerRailItemProps = SharedStyleProps & {
-    /** Custom classes for default style overrides */
-    classes?: DrawerRailItemClasses;
+export type DrawerRailItemProps = SharedStyleProps &
+    MuiButtonBaseProps & {
```
Comment (joebochill) [inline components/src/core/Drawer/DrawerRailItem/DrawerRailItem.tsx:46]: This is a little weird. Should we be deprecating the ButtonBaseProps prop below? How is that currently being used?

**Comment by bkarambe** [comment]
> do you have an example of how to use this for rail?

Please checkout [feature/rail-item-style-example](https://github.com/brightlayer-ui/react-showcase-demo/tree/feature/rail-item-style-example) branch in react-showcase-demo

File: components/src/core/Drawer/DrawerRailItem/DrawerRailItem.tsx (hunk 1 above)
Comment (bkarambe) [reply components/src/core/Drawer/DrawerRailItem/DrawerRailItem.tsx:46]: Currently, it is not accepting style in DrawerNavItem

**Review by jeffvg — APPROVED** [review]
style & sx seems good

File: components/src/core/Drawer/DrawerRailItem/DrawerRailItem.tsx (hunk 1 above)
Comment (daileytj) [reply components/src/core/Drawer/DrawerRailItem/DrawerRailItem.tsx:46]: @joebochill @bkarambe did we ever clear this up? What's the approach we want to take here?

Comment (bkarambe) [reply components/src/core/Drawer/DrawerRailItem/DrawerRailItem.tsx:46]: Yes, we will be using MuiButtonBaseProps in DrawerRailItemProps

**Review by daileytj — APPROVED** [review]

**Pull request merged**